pyccel/ast/variable.py

```diff
@@ -579,20 +585,69 @@ class TupleVariable(Variable):
     >>> n
     n
     """
-    __slots__ = ('_vars','_inconsistent_shape','_is_homogeneous')
+    __slots__ = ()
+
+class HomogeneousTupleVariable(TupleVariable):
+
+    """Represents a tuple variable in the code.
+
+    Parameters
+    ----------
+    arg_vars: Iterable
+        Multiple variables contained within the tuple
+
+    Examples
+    --------
+    >>> from pyccel.ast.core import TupleVariable, Variable
+    >>> v1 = Variable('int','v1')
+    >>> v2 = Variable('bool','v2')
+    >>> n  = TupleVariable([v1, v2],'n')
+    >>> n
+    n
+    """
+    __slots__ = ()
+    is_homogeneous = True
```

Comment:
It can be useful. E.g [here](https://github.com/pyccel/pyccel/pull/842/files#diff-9ba0c438140113a543a41e16000b939989e841d4fc6a6a4112b5c8e5d8807524R1646)
The condition:
`isinstance(a, (TupleVariable, PythonTuple)) and a.is_homogeneous`
would have to be written as:
`(isinstance(a, PythonTuple) and a.is_homogeneous) or isinstance(a, HomogeneousTupleVariable)`

Reply:
I see, you want to keep it because it is a member of the `PythonTuple` class

Reply:
Instances of `PythonTuple` and `TupleVariable` can be used interchangeably most of the time so it makes sense for them to have a certain number of properties in common. It is also used [here](

That said, I think the link above is the only place where I use this property and I've just realised there is a simpler test that should replace it here. So the property can maybe be removed